resources/security_group.rb

```diff
@@ -0,0 +1,255 @@
+#
+# Cookbook:: aws
+# Resource:: security_group
+#
+# Technically managing the security group and the ingress/egress
+# rules are unique API calls.  It's much easier to handle them in
+# a single provider, partially to limit the number of API calls.
+# Mostly because describing a security group returns the entire object
+# which we can use to compare against our object
+
+resource_name :security_group
+provides :aws_security_group
+
+# => Define the Resource Properties
+property :security_group_name, String, name_property: true
+property :description, String, default: 'created by chef'
+property :vpc_id, String, required: true
+
+# Ingress/Egress rules
+property :ip_permissions, Array, default: []
```

Comment:
can we add an alias for `ip_permissions_ingress`? I get this matches the aws sdk and we should follow that but we should probably make it so that implementers can opt for more explicit and readable code.

Reply:
alias added, thanks